### ISSUE TITLE
Code coverage reports generates invalid report when having 100% coverage

### ISSUE THREAD

**Issue by mnunezdm** [forcedotcom/cli#1553]
### Summary

Hi! first of all very huge thank you for implementing the functionality for generating the test & code coverage report. We currently have an in house script to generate this reports from a json from a deployment but its much better to have it directly integrated in the cli

What I found is that, when you generate a code coverage report for a fully covered class, in the lines list, instead of having a hit for each line, you have one entry for line number infinity. In my pipeline (Azure Pipelines) this breaks the report generation

```xml
<class name="XXX_class" filename="no-map/XXX_class" line-rate="1" branch-rate="1">
    <methods>
    </methods>
    <lines>
        <line number="Infinity" hits="1" branch="false"/>
    </lines>
</class>
```

In my scenario, deleting this line `<line number="Infinity" hits="1" branch="false"/>` fixes the report generation but the report generated is "invalid"

![image](https://user-images.githubusercontent.com/10410852/170838112-a82846d1-a804-4fbb-ab9b-b537e177c3e2.png)

### Steps To Reproduce:

Deploy dreamhouse project running tests and generating code coverage report

```
> sfdx force:source:deploy -p force-app --runtests="TestPropertyController,TestSampleDataController,GeocodingServiceTest" --testlevel=RunSpecifiedTests --coverageformatters cobertura
```

### Expected result

I assume that the correct result would be to generate a line tag for each covered line, but I don't know how can this be accomplished as the deployment json does not return any line coverage info if the file has 100% code coverage

### Actual result

```xml
<class name="GeocodingService" filename="no-map/GeocodingService" line-rate="1" branch-rate="1">
    <methods></methods>
    <lines>
        <line number="Infinity" hits="1" branch="false" />
    </lines>
</class>
<class name="PagedResult" filename="no-map/PagedResult" line-rate="1" branch-rate="1">
    <methods></methods>
    <lines>
        <line number="Infinity" hits="1" branch="false" />
    </lines>
</class>
```

### System Information

```json
{
        "cliVersion": "sfdx-cli/7.153.0",
        "architecture": "darwin-x64",
        "nodeVersion": "node-v16.15.0",
        "pluginVersions": [
                "@oclif/plugin-autocomplete 0.3.0 (core)",
                "@oclif/plugin-commands 1.3.0 (core)",
                "@oclif/plugin-help 3.3.1 (core)",
                "@oclif/plugin-not-found 1.2.6 (core)",
                "@oclif/plugin-plugins 1.10.11 (core)",
                "@oclif/plugin-update 1.5.0 (core)",
                "@oclif/plugin-warn-if-update-available 1.7.3 (core)",
                "@oclif/plugin-which 1.0.4 (core)",
                "@salesforce/sfdx-plugin-lwc-test 0.1.7 (core)",
                "alias 2.0.1 (core)",
                "apex 0.13.0 (core)",
                "auth 2.0.3 (core)",
                "community 1.1.5 (core)",
                "config 1.4.7 (core)",
                "custom-metadata 1.1.0 (core)",
                "data 2.0.2 (core)",
                "generator 2.0.1 (core)",
                "info 2.0.1 (core)",
                "limits 2.0.1 (core)",
                "org 1.13.2 (core)",
                "salesforce-alm 54.4.0 (core)",
                "schema 2.1.1 (core)",
                "sfdx-cli 7.153.0 (core)",
                "signups 1.1.2 (core)",
                "source 1.10.1 (core)",
                "telemetry 2.0.0 (core)",
                "templates 54.8.0 (core)",
                "trust 1.1.0 (core)",
                "user 2.0.2 (core)"
        ],
        "osVersion": "Darwin 21.4.0"
}
```

### Additional information

_Feel free to attach a screenshot_.

**Comment by uip-robot-zz** [comment]
This issue has been linked to a new work item: [W-11221748](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00000y9uvUYAQ/view)

**Comment by peternhale** [comment]
@mnunezdm we have identified the issue and are working on a fix. Thanks for reporting the issue.

**Comment by cristiand391** [comment]
Hey @mnunezdm, the fix for this was included in sfdx v7.153.1 which is currently latest, let us know if it works for you, thanks!

cc @WillieRuemmele

**Comment by mnunezdm** [comment]
Hello @cristiand391 I've just checked and seems that this fixes the error

Thx!

Feel free to close the issue!
